dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,576 @@
+            );
+        }
+
+        //The Service does exist and so it does the secrets.
+        final Response siteIntegrationsResponse = serviceIntegrationResource
+                .getServiceIntegrationByKey(request, response, serviceKey);
+        Assert.assertEquals(HttpStatus.SC_OK, siteIntegrationsResponse.getStatus());
+        final ResponseEntityView responseEntityView = (ResponseEntityView) siteIntegrationsResponse
+                .getEntity();
+        final ServiceIntegrationView serviceIntegrationView = (ServiceIntegrationView) responseEntityView
+                .getEntity();
+
+        Assert.assertEquals(max, serviceIntegrationView.getConfigurationsCount());
+
+        for (final String siteId : sites) {
+            final Response detailedIntegrationResponse = serviceIntegrationResource
+                    .getDetailedServiceIntegration(request, response, serviceKey, siteId);
+            Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+            final ResponseEntityView responseEntityView3 = (ResponseEntityView) detailedIntegrationResponse
+                    .getEntity();
+            final ServiceIntegrationView serviceIntegrationDetailedView = (ServiceIntegrationView) responseEntityView3
+                    .getEntity();
+            Assert.assertNotNull(serviceIntegrationDetailedView.getSites());
+            Assert.assertFalse(serviceIntegrationDetailedView.getSites().isEmpty());
+            final Map<String, Secret> secrets = serviceIntegrationDetailedView.getSites().get(0)
+                    .getSecrets();
+            for (Entry<String, Secret> secretEntry : secrets.entrySet()) {
+                final String key = secretEntry.getKey();
+                final Secret secret = secretEntry.getValue();
+                final Param originalParam = initialParamsMap.get(key);
+                if (secret.isHidden()) {
+                    Assert.assertEquals(ServiceIntegrationHelper.PROTECTED_HIDDEN_SECRET,
+                            new String(secret.getValue()));
+                } else {
+                    Assert.assertEquals(originalParam.getString(), new String(secret.getValue()));
+                }
+            }
+        }
+    }
+
+
+    @Test
+    public void Test_Service_Key_Casing() throws IOException {
+
+        final Map<String, Param> initialParamsMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true));
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        long time = System.currentTimeMillis();
+
+        final String serviceKey1 = String.format("all_lower_case_not_too_short_prefix_%d", time);
+        final String fileName = String.format("%s.yml", serviceKey1);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey1,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)